### PR TITLE
Feature/intake form document upload

### PR DESCRIPTION
**Pull request by loshjarson**
Why?
Allow the ability to upload documents for clients.

What work was done?
Added document upload component with accompanying reducer state and actions.

Loom video: https://www.loom.com/share/5e7962e91a1841c99c5b8051225c4935
Trello card: https://trello.com/c/l0SvH9kT

### CODE REVIEW

File: src/components/common/ClientIntakeForm/IntakeForms/DocumentUpload.js [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+          )}
+        </Space>
+      ),
+    },
+  ];
+
+  useEffect(() => {
+    if (props.client.id !== 0) {
+      axios
+        .get(`http://localhost:8000/api/documents/${props.client.id}`)
```
Comment (tinomen) [inline src/components/common/ClientIntakeForm/IntakeForms/DocumentUpload.js:100]: We should put the root api url into an env var. (.env file) `http://localhost:8000` won't work when deployed to production. Enviornment variables are so helpful for values that change based on the enviornment.

File: src/components/common/ClientIntakeForm/IntakeForms/DocumentUpload.js [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+        })
+        .catch(error => {
+          alert('ERROR ' + error);
+        });
+    }
+  }, [props.client]);
+
+  const handleDelete = file => {
+    axios
+      .post(`http://localhost:8000/api/documents/${props.client.id}`, file)
```
Comment (tinomen) [inline src/components/common/ClientIntakeForm/IntakeForms/DocumentUpload.js:114]: Same as above, lets move the root API URL into an env var.

File: src/components/common/ClientIntakeForm/IntakeForms/DocumentUpload.js [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+    {
+      title: 'Type',
+      dataIndex: 'documentType',
+      key: 'type',
+      filters: documentTypeFilter,
+    },
+    {
+      title: 'Options',
+      key: 'options',
+      render: (text, record, i) => (
```
Comment (tinomen) [inline src/components/common/ClientIntakeForm/IntakeForms/DocumentUpload.js:73]: Great idea to add an icon button for the delete function

File: src/components/common/ClientIntakeForm/IntakeForms/DocumentUpload.js [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+  };
+
+  const upload = async e => {
+    console.log(e.file.name, { ...e.file });
+
+    setDisabled(true);
+    let file = e.file;
+    let fileParts = e.file.name.split('.');
+    let fileName = fileParts[0];
+    let fileType = fileParts[1];
```
Comment (tinomen) [inline src/components/common/ClientIntakeForm/IntakeForms/DocumentUpload.js:130]: What happens if someone drags a file onto the component but the document type wasn't chosen?

File: src/components/common/ClientIntakeForm/IntakeForms/DocumentUpload.js [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+              name: fileName,
+              success: true,
+              url: url,
+              documentType: documentType,
+              extension: fileType,
+            };
+            props.addDocument([documentType, documentObj]);
+            axios
+              .post(
+                `http://localhost:8000/api/documents/${props.client.id}/update`,
```
Comment (tinomen) [inline src/components/common/ClientIntakeForm/IntakeForms/DocumentUpload.js:158]: api url env var

File: src/components/common/ClientIntakeForm/IntakeForms/DocumentUpload.js [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+  },
+  {
+    label: 'Birth certificates for children in household',
+    value: 'birth_cert_for_children',
+  },
+  { label: 'Children Enrolled in School', value: 'child_enrolled_school' },
+  { label: 'Childcare', value: 'childcare' },
+];
+
+const documentTypeFilter = [
```
Comment (tinomen) [inline src/components/common/ClientIntakeForm/IntakeForms/DocumentUpload.js:43]: Can we make the `text` property have spaces so it is cleaner for the user.

File: src/components/common/ClientIntakeForm/IntakeForms/DocumentUpload.js (hunk 4 above)
Comment (tinomen) [reply src/components/common/ClientIntakeForm/IntakeForms/DocumentUpload.js:130]: ok, I found the disabled property on the dragger. I think this interaction is complicated enough to warrant some comments on the onSelect function.

File: src/components/common/ClientIntakeForm/IntakeForms/DocumentUpload.js [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+          .catch(error => {
+            alert('ERROR ' + error);
+          });
+      })
+      .catch(error => {
+        alert('ERROR ' + error);
+      });
+  };
+
+  const onSelect = e => {
```
Comment (tinomen) [inline src/components/common/ClientIntakeForm/IntakeForms/DocumentUpload.js:177]: Do you think this function would be more clear to other devs if it's name was specific to the DOM element and event?

**Review by tinomen — CHANGES_REQUESTED** [review]
Great work putting everything together on this feature.

File: src/components/common/ClientIntakeForm/IntakeForms/DocumentUpload.js (hunk 3 above)
Comment (loshjarson) [reply src/components/common/ClientIntakeForm/IntakeForms/DocumentUpload.js:73]: Thanks!

File: src/components/common/ClientIntakeForm/IntakeForms/DocumentUpload.js (hunk 4 above)
Comment (loshjarson) [reply src/components/common/ClientIntakeForm/IntakeForms/DocumentUpload.js:130]: I added a brief comment, let me know if I should further elaborate

**Pull request merged**